core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -198,7 +200,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         /* generate brokerId */
         val (brokerId, initialOfflineDirs) = getBrokerIdAndOfflineDirs
         config.brokerId = brokerId
-        this.logIdent = "[Kafka Server " + config.brokerId + "], "
+        logContext = new LogContext("[Kafka Server " + config.brokerId + "] ")
```

Comment:
Similarly, this can be changed to `new LogContext("[KafkaServer id=${config.brokerId}] ")`